velox/functions/prestosql/Arithmetic.h

```diff
@@ -495,5 +495,41 @@ struct TruncateFunction {
   }
 };
 
+template <typename T>
+struct BinomialCDF {
+  // The computation of binomial cumulative distribution follows the algorithm
+  // here: https://mathworld.wolfram.com/BinomialDistribution.html
+
+  double choose(int64_t n, int64_t k) {
+    if (k == 0) {
+      return 1;
+    } else {
+      return (n * choose(n - 1, k - 1)) / k;
+    }
+  }
+
+  FOLLY_ALWAYS_INLINE void
+  call(double& result, int64_t numOfTrials, double successProb, int64_t value) {
+    VELOX_USER_CHECK_GT(
+        numOfTrials, 0, "Number of trials must be greater than 0");
+    VELOX_USER_CHECK_GE(value, 0, "Value must be a positive integer");
+    VELOX_USER_CHECK_GE(
+        numOfTrials,
+        value,
+        "Number of trials must be greater than or equal to value");
+    VELOX_USER_CHECK(
+        (successProb >= 0) && (successProb <= 1),
+        "Success probability must be real value in [0, 1]");
+
+    double failureProb = 1 - successProb;
```

Comment:
Is it possible to use binomial distribution from the boost libraries for this logic. e.g. https://www.boost.org/doc/libs/1_67_0/boost/math/distributions/binomial.hpp has a cdf funtion.

Reply:
Changed to boost library.